bob/src/api/http.rs

```diff
@@ -236,6 +245,29 @@ async fn status(bob: &State<BobServer>) -> Json<Node> {
     Json(node)
 }
 
+#[get("/status/space")]
+async fn get_space_info(bob: &State<BobServer>) -> Result<Json<SpaceInfo>, StatusExt> {
+    let snapshot = bob.metrics().read().await;
+    let total_space = snapshot.gauges_map.get(TOTAL_SPACE).map(|g| g.value);
+    let used_space = snapshot.gauges_map.get(USED_SPACE).map(|g| g.value);
+    let free_space = snapshot.gauges_map.get(FREE_SPACE).map(|g| g.value);
+    total_space
+        .zip(used_space)
+        .zip(free_space)
```

Comment:
Why do you use `zip` on scalar values? It would be easier to construct `SpaceInfo` directly from `total_space`, `used_space` and `free_space` values

Reply:
They are `Option`s, this way if any of them is `None` the whole result will be none, no need to unwrap/check individual values